bin/citrea/src/cli.rs

```diff
@@ -59,6 +61,12 @@ pub(crate) struct Args {
     /// Logging verbosity
     #[arg(long, short = 'q', action)]
     pub(crate) quiet: bool,
+
+    /// Optionally restore DB state from a backup before starting the node.
+    /// If provided, databases are restored from the specified backup path
+    /// before any services are initialized.
+    #[arg(long)]
+    pub(crate) restore_db: Option<PathBuf>,
```

Comment:
As part of https://github.com/chainwayxyz/citrea/pull/1796, i am working on `citrea-cli` which provides a bunch of subcommands that we can use to operate the node's storage among potentially other stuff without having to put these things in the node's binary.

I believe this can be a subcommand `restore-backup` or something of the sort which could trigger restoring the DB from backup, rather than having to pass this flag while running the node itself. IMO, not a good design / use case.

Reply:
I don't think so... because there will be some disparity between some in-memory data and what's been written to the database while the node is running. 
Your code restores at the beginning of the binary execution which can be done in a step before running the node which is by invoking `citrea-cli restore-backup ... etc`.

However, i think your PR will get merged before mine so we can address this later.